src/rust/ide/lib/view/src/graph_editor/component/breadcrumbs.rs

```diff
@@ -0,0 +1,555 @@
+    /// be popped or a list of `LocalCall`s identifying the breadcrumbs we need to push.
+    fn select_breadcrumb(&self, index:usize) -> (usize,Vec<Option<LocalCall>>) {
+        info!(self.logger,"Selecting breadcrumb #{index}.");
+        let current_index = self.current_index.get();
+        match index.cmp(&current_index) {
+            Ordering::Less    => (current_index - index, default()),
+            Ordering::Equal   => default(),
+            Ordering::Greater => {
+                let mut local_calls = Vec::new();
+                for index in current_index..index {
+                    let info = self.breadcrumbs.borrow().get(index).map(|breadcrumb| {
+                        let definition = breadcrumb.info.method_pointer.clone();
+                        let call       = breadcrumb.info.expression_id;
+                        LocalCall{call,definition}
+                    }).as_ref().cloned();
+                    if info.is_some() {
+                        local_calls.push(info);
+                    } else {
+                        error!(self.logger, "LocalCall info is not present.");
+                        self.remove_breadcrumbs_history_beginning_from(index);
+                        break;
+                    }
+                }
+                (default(),local_calls)
+            },
+        }
+    }
+
+    fn push_breadcrumb(&self, local_call:&Option<LocalCall>) -> Option<(usize,usize)> {
+        local_call.as_ref().map(|local_call| {
+            let method_pointer = &local_call.definition;
+            let expression_id  = &local_call.call;
+            let old_index      = self.current_index.get();
+            let new_index      = old_index + 1;
+
+            let breadcrumb_exists =
+                self.breadcrumbs.borrow_mut().get(old_index).contains_if(|breadcrumb| {
+                    breadcrumb.info.expression_id == *expression_id
+                });
+
+            if breadcrumb_exists {
+                debug!(self.logger, "Entering an existing {method_pointer.name} breadcrumb.");
+            } else {
+                debug!(self.logger, "Creating a new {method_pointer.name} breadcrumb.");
+                self.remove_breadcrumbs_history_beginning_from(self.current_index.get());
+                let breadcrumb       = Breadcrumb::new(&self.scene, method_pointer, expression_id);
+                let network          = &breadcrumb.frp.network;
+                let breadcrumb_index = new_index;
+                let frp_inputs       = &self.frp_inputs;
+
+
+                // === User Interaction ===
+
+                frp::extend! { network
+                    eval_ breadcrumb.frp.outputs.clicked(
+                        frp_inputs.select_breadcrumb.emit(breadcrumb_index);
+                    );
+                }
+
+                info!(self.logger, "Pushing {breadcrumb.info.method_pointer.name} breadcrumb.");
```

Comment:
Is this info so important that user should see it without debug mode? Please remove all similar logging from this PR. Just use `debug!` instead.